package assertions_exeptions;

public class exo1  {
    public void f(int a) {   //throws IllegalArgumentException {
    	if(a<2) throw new IllegalArgumentException("Probleme valeur param");
    	System.out.println("a = " +a);
    }


    public static void main(String[] args) {
        Test t = new Test();
        Scanner sc = new Scanner(System.in);
        boolean nonfin = false;
        do {
            try {
                int val;
                nonfin = false;
                System.out.print("Entrer un nombre :");
                val = sc.nextInt();
                t.f(val);
            } catch (MonException ee) {
                System.out.println("Erreur : " + ee);
                nonfin = true;
            }
        } while (nonfin);
    }
}

